Panic test time out from addons/controller

**Bug description**

From main branch, I consistently see test time out error from local `go test` or Github CI. With more `xx_test.go` being added, it needs more time to finish, however the default timeout is 10m.

```
panic: test timed out after 10m0s

goroutine 27624 [running]:
testing.(*M).startAlarm.func1()
        /usr/local/Cellar/go/1.17.6/libexec/src/testing/testing.go:1788 +0x8e
created by time.goFunc
        /usr/local/Cellar/go/1.17.6/libexec/src/time/sleep.go:180 +0x31

goroutine 1 [chan receive, 10 minutes]:
testing.(*T).Run(0xc000602820, {0x276f81e, 0x330c79f41c5df}, 0x2834d28)
        /usr/local/Cellar/go/1.17.6/libexec/src/testing/testing.go:1307 +0x375
testing.runTests.func1(0xc000445080)
        /usr/local/Cellar/go/1.17.6/libexec/src/testing/testing.go:1598 +0x6e
testing.tRunner(0xc000602820, 0xc0007b7d18)
        /usr/local/Cellar/go/1.17.6/libexec/src/testing/testing.go:1259 +0x102
testing.runTests(0xc0001c0080, {0x38e2480, 0x1, 0x1}, {0x0, 0x1027689b2, 0x391d9a0})

...

goroutine 21009 [select]:
github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks.(*WebhookTLS).manageTLSCertificates(0xc0001e28c0, 0xc000be1e50)
        /Users/gdaniel/go/src/github.com/danniel1205/tanzu-framework/addons/vendor/github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks/webhooks.go:103 +0x95
created by github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks.(*WebhookTLS).ManageCertificates
        /Users/gdaniel/go/src/github.com/danniel1205/tanzu-framework/addons/vendor/github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks/webhooks.go:96 +0x94

goroutine 20989 [select]:
github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks.(*WebhookTLS).manageTLSCertificates(0xc0018e0f00, 0xc00122a460)
        /Users/gdaniel/go/src/github.com/danniel1205/tanzu-framework/addons/vendor/github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks/webhooks.go:103 +0x95
created by github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks.(*WebhookTLS).ManageCertificates
        /Users/gdaniel/go/src/github.com/danniel1205/tanzu-framework/addons/vendor/github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks/webhooks.go:96 +0x94
FAIL    github.com/vmware-tanzu/tanzu-framework/addons/controllers      600.572s
FAIL
```

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**

Local `go test` or Github CI passes on tests

**Steps to reproduce the bug**

- Run `go test -v github.com/vmware-tanzu/tanzu-framework/addons/controllers`
- Test times out and fails

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

--- a/addons/Makefile
+++ b/addons/Makefile
@@ -51,7 +51,7 @@ help: ## Display this help
 .PHONY: test
 test: fmt vet template-tests ## Run Tests
 	$(MAKE) kubebuilder -C ${ROOT_DIR}$(TOOLS_DIR)
-	KUBEBUILDER_ASSETS=${ROOT_DIR}$(KUBEBUILDER_BIN_DIR) go test ./... -coverprofile cover.out -v 2
+	KUBEBUILDER_ASSETS=${ROOT_DIR}$(KUBEBUILDER_BIN_DIR) go test ./... -coverprofile cover.out -v 2 -timeout 15m
 	$(MAKE) test -C pinniped/post-deploy
 
 .PHONY: test-verbose

